compat/src/react-18-hooks.js

```diff
@@ -0,0 +1,54 @@
+import { useState, useLayoutEffect, useEffect } from 'preact/hooks';
+
+/**
+ * Asynchronously schedule a callback
+ * @type {(cb: () => void) => void}
+ */
+/* istanbul ignore next */
+// Note the following line isn't tree-shaken by rollup cuz of rollup/rollup#2566
+const defer =
```

Comment:
I think `requestAnimationFrame` is more correct than using `Promise.resolve().then` (or `queueMicrotask` which should be more efficient).
From my understanding, the idea is to defer some updates so that more urgent ones can take place. But microtasks have a very high priority.